### PR TITLE
feature(barman): Add additional conf options

### PR DESCRIPTION
**Pull request by ggggut**
# Description

Add additional Values for configuring container `volumeMounts` and deployment `volumes`.
This will be useful if you want to have pre/post scripts in Barman (configmaps).  
It templates the whole thing with toYaml. 

Furthermore I added `additionalConfiguration` to make all parameters from https://docs.pgbarman.org/release/2.19/barman.5.html available for configuring your barman instance. 
This will take a string + multiline string. So configuration would look like:

````
# this
  backups:
      additionalConfiguration: |
        pre_backup_script = "/path/to/your/script.py"
        minimum_redundancy = 2

# or this
  backups:
      additionalConfiguration: "minimum_redundancy = 2"
````
# Issues

# Checklist

* [x] This PR contains a description of the changes I'm making
* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run`
* [ ] I documented any high-level concepts I'm introducing in `docs/`
* [ ] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/barman/templates/configmap-barman-backups.yaml [hunk 1]
```diff
@@ -25,4 +25,7 @@ data:
       {{- if hasKey . "retentionPolicy" }}
       retention_policy = {{ .retentionPolicy }}
       {{- end }}
+      {{- if hasKey . "additionalConfiguration" }}
+{{ .additionalConfiguration | indent 6 }}
```
Comment (hairmare) [inline charts/barman/templates/configmap-barman-backups.yaml:29]:
```suggestion
      {{ .additionalConfiguration | nindent 6 }}
```

**Review by hairmare — CHANGES_REQUESTED** [review]
one tiny nit: please use `nindent` instead of `indent` for consistency

other than that, this LGTM from my pov

File: charts/barman/templates/configmap-barman-backups.yaml (hunk 1 above)
Comment (ggggut) [reply charts/barman/templates/configmap-barman-backups.yaml]: Fixed 👍

**Comment by hairmare** [comment]
Thanks very much for the contribution!

**Pull request merged**